TWLight/users/signals.py

```diff
@@ -1,4 +1,41 @@
-import django.dispatch
+from datetime import timedelta
+from django.dispatch import receiver, Signal
+from django.db.models.signals import post_save
+from TWLight.users.models import Authorization
+from .models import Partner, Stream
+
 
 class Notice(object):
-    user_renewal_notice = django.dispatch.Signal(providing_args=['user_wp_username', 'user_email', 'user_lang', 'partner_name', 'partner_link'])
+    user_renewal_notice = Signal(providing_args=['user_wp_username', 'user_email', 'user_lang', 'partner_name', 'partner_link'])
+
+
+
+@receiver(post_save, sender=Partner)
+@receiver(post_save, sender=Stream)
+def update_partner_authorization_expiry(sender, instance, **kwargs):
+    """
+    Updates missing expiration dates upon resource updates.
+    Mostly cribbed from
+    TWLight.applications.models.post_receive_commit
+    and
+    TWLight.users.management.commands.authorization_backfill
+    Could factor out the common code.
+    """
+
+    print(sender)
+    if sender == Partner:
+        partner = instance
+    elif sender == Stream:
+        partner = instance.partner
+
+    if partner.account_length or partner.requested_access_duration:
+        authorizations = Authorization.objects.filter(partner=partner)
```

Comment:
Oh I've just noticed we check for `not authorization.date_expires` below. Could it make sense to just filter on that in the first place?

Reply:
Yeah, I didn't do it that way, because I thought there might be situations in which we needed to update existing expiration dates. I meant to pose that question in the pr body, but instead just posted it empty. If we really only care about empty expiration dates, then filtering the query set would definitely be more performant

Reply:
I think if there's an existing expiration date (only currently the case in a handful of places if I recall) then if anything I don't think users would be surprised that the same date remains after switching.